core/storage/btree.rs

```diff
@@ -1715,16 +1714,24 @@ impl BTreeCursor {
                         new_page_sizes[i + 1] -= size_of_cell_to_remove_from_right;
                     }
 
-                    let we_still_need_another_page =
+                    // Check if this page contains up to the last cell. If this happens it means we really just need up to this page.
+                    // Let's update the number of new pages to be up to this page (i+1)
+                    let page_completes_all_cells =
                         cell_array.number_of_cells_per_page[i] >= cell_array.cells.len() as u16;
-                    if we_still_need_another_page {
+                    if page_completes_all_cells {
```

Comment:
I'm confused about how `we_still_need_another_page` got renamed to `page_completes_all_cells`, signifying (as far as I understand), that we don't need another page?

Reply:
Basically this last page completed the list of cells, meaning we don't need another page to move cells to because this one contains the last cell.

Reply:
but why was it `we_still_need_another_page` before, if the meaning is `we_dont_actually_need_another_page_anymore`

Reply:
i think `is_last_needed_page` sounds good for the meaning you are after:

> Basically this last page completed the list of cells, meaning we don't need another page to move cells to because this one contains the last cell.

Reply:
Before it was called `we_still_need_another_page` because I was stupid and read the code wrong so I assumed it did another thing XD. This part only checks if we don't need more pages, and the first part of the loops will add more pages if we _need_ to move cells to the right.